### DISCUSSION TITLE
Performance of back-to-back ts-eslint runs

### DISCUSSION THREAD

**Discussion by innermatrix** [typescript-eslint/typescript-eslint#9046] [Technical Discussions]
### Body

I find myself in a situation where I want to run `eslint` twice back-to-back in circumstances where the source will typically remain unchanged between two runs (think: pre-commit hook or CI pipeline). I use type-aware rules, and I know that due to the current architecture of `eslint` and `ts-eslint` I need to pay the penalty of constructing all type information before a single file can be linted. 

However…

Is there a way that I can avoid paying the penalty twice when making two back-to-back `eslint` runs (if the source is unchanged)? As I understand it, the `eslint --cache` option is unhelpful here because the presence of type-aware rules busts the `eslint` cache. 

(The use case here is that I would like to use `eslint` for checking rules that have been adopted by my team, and `betterer` for checking rules that are being incrementally adopted, and `betterer` is challenging to configure as a runner for all `eslint` rules due to its own architectural decisions.)

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per our discussion guidelines we don't use discussions for support.
https://typescript-eslint.io/contributing/discussions

Consider using discord or similar. The guidelines contain helpful suggestions and links.
